dvc/dependency/repo.py

```diff
@@ -46,13 +47,22 @@ def _make_repo(self, **overrides):
             yield repo
 
     def status(self):
-        with self._make_repo() as repo:
-            current = repo.find_out_by_relpath(self.def_path).info
-
-        with self._make_repo(rev_lock=None) as repo:
-            updated = repo.find_out_by_relpath(self.def_path).info
-
-        if current != updated:
+        with self._make_repo() as repo, self._make_repo(
+            rev_lock=None
+        ) as updated_repo:
+            try:
+                current = repo.find_out_by_relpath(self.def_path).info
+                updated = updated_repo.find_out_by_relpath(self.def_path).info
+
+                has_changed = current != updated
+            except OutputNotFoundError:
```

Comment:
Hm, there is also an interesting case, when a file/dir was git-tracked before but then became dvc-tracked or vice versa. But the thingy with get_checksum should make those comparable, I think.